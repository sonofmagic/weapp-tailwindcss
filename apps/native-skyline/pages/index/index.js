import { cityData } from './data'
import { add } from 'lodash'
Component({
  data: {
    list: [],
  },
  lifetimes: {
    attached() {
      const cities = cityData
      // 按拼音排序
      cities.sort((c1, c2) => {
        const pinyin1 = c1.pinyin.join('')
        const pinyin2 = c2.pinyin.join('')
        return pinyin1.localeCompare(pinyin2)
      })
      // 添加首字母
      const map = new Map()
      for (const city of cities) {
        const alpha = city.pinyin[0].charAt(0).toUpperCase()
        if (!map.has(alpha)) map.set(alpha, [])
        map.get(alpha).push({ name: city.fullname })
      }

      const keys = []
      for (const key of map.keys()) {
        keys.push(key)
      }
      keys.sort()

      const list = []
      for (const key of keys) {
        list.push({
          alpha: key,
          subItems: map.get(key)
        })
      }

      console.log('address-book list:', list)
      this.setData({ list })
      console.log(add(1,3));
    },
  },
})
