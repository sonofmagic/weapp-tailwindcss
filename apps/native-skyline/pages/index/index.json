{
  "usingComponents": {
    "navigation-bar": "../../components/navigation-bar",
    "address-book": "../../components/address-book"
  },
  "disableScroll": true,
  "navigationStyle": "custom"
}