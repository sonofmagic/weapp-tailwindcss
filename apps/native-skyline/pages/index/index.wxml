<navigation-bar title="通讯录" back="{{true}}"></navigation-bar>
<view class="w-screen grow overflow-hidden">
  <address-book list="{{list}}">
    <view class="text-black/90">
      <view class="p-10">
        <view class="page__title">Address Book</view>
        <view class="page__desc">类通讯录列表</view>
        <view class="i-mdi-home bg-yellow-600 w-4 h-4"></view>
      </view>
      <view class="pb-10">
      </view>
    </view>
  </address-book>
</view>
