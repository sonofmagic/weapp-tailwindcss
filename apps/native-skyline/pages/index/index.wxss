page {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  height: 100vh;
  background-color: #f7f7f7;
}

.page__title {
  @apply text-center text-xl;
}

.page__desc {
  @apply text-center text-sm;
  margin-top: 5px;
  color: rgba(0, 0, 0, 0.5);
}

.cell {
  height: 50px;
  justify-content: center;
  align-items: center;
}
