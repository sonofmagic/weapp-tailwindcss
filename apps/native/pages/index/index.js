// index.js
// 获取应用实例
const app = getApp()

Page({
  data: {
    motto: 'Hello World',
    userInfo: {},
    hasUserInfo: false,
    canIUse: wx.canIUse('button.open-type.getUserInfo'),
    imgUrls: null,
    indicatorDots: true,
    autoplay: true,
    interval: 5000,
    duration: 1000,
    list: [],
    page: 1
  },
  // 事件处理函数
  bindViewTap: function () {
    wx.navigateTo({
      url: '../logs/logs'
    })
  },
  addCart(data) {
    const item = data.currentTarget.dataset.item
  },
  imgsc: function (e) {
    wx.chooseImage({
      count: 1, // 默认9
      sizeType: ['original', 'compressed'], // 可以指定是原图还是压缩图，默认二者都有
      sourceType: ['album', 'camera'], // 可以指定来源是相册还是相机，默认二者都有
      success: function (res) {
        // 返回选定照片的本地文件路径列表，tempFilePath可以作为img标签的src属性显示图片
        const tempFilePaths = res.tempFilePaths
      }
    })
  },
  lower: function (e) {
    console.log(e)
    this.getList()
  },
  getList: function () {},
  onLoad: function () {
    const app = getApp()
    this.getList()

    if (app.globalData.userInfo) {
      this.setData({
        userInfo: app.globalData.userInfo,
        hasUserInfo: true
      })
    } else if (this.data.canIUse) {
      // 由于 getUserInfo 是网络请求，可能会在 Page.onLoad 之后才返回
      // 所以此处加入 callback 以防止这种情况
      app.userInfoReadyCallback = (res) => {
        this.setData({
          userInfo: res.userInfo,
          hasUserInfo: true
        })
      }
    } else {
      // 在没有 open-type=getUserInfo 版本的兼容处理
      wx.getUserInfo({
        success: (res) => {
          app.globalData.userInfo = res.userInfo
          this.setData({
            userInfo: res.userInfo,
            hasUserInfo: true
          })
        }
      })
    }
  },
  changeIndicatorDots: function (e) {
    this.setData({
      indicatorDots: !this.data.indicatorDots
    })
  },
  changeAutoplay: function (e) {
    this.setData({
      autoplay: !this.data.autoplay
    })
  },
  intervalChange: function (e) {
    this.setData({
      interval: e.detail.value
    })
  },
  durationChange: function (e) {
    this.setData({
      duration: e.detail.value
    })
  },
  getUserInfo: function (e) {
    app.globalData.userInfo = e.detail.userInfo
    this.setData({
      userInfo: e.detail.userInfo,
      hasUserInfo: true
    })
  }
})
