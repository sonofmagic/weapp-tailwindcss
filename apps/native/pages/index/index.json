{
  "navigationBarTitleText": "首页",
  "enablePullDownRefresh":true,
  "backgroundTextStyle":"dark"
}