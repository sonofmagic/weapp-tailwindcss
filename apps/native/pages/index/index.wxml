<!-- index.wxml -->
<scroll-view class="scrollarea" scroll-y type="list">
  <view class="container">
    <view class="userinfo">
      <block wx:if="{{canIUseNicknameComp && !hasUserInfo}}">
        <button open-type="chooseAvatar" bind:chooseavatar="onChooseAvatar">
          <image class="w-[100px] h-[100px]" src="{{userInfo.avatarUrl}}"></image>
        </button>
        <view class="nickname-wrapper">
          <text class="nickname-label">昵称</text>
          <input type="nickname" class="nickname-input" placeholder="请输入昵称" bind:change="onInputChange" />
        </view>
        <view class="text-[#432432]">{{ num }}</view>
      </block>
      <block wx:elif="{{!hasUserInfo}}">
        <button wx:if="{{canIUseGetUserProfile}}" bindtap="getUserProfile">获取头像昵称</button>
        <view wx:else>请使用2.10.4及以上版本基础库</view>
      </block>
      <block wx:else>
        <image bindtap="bindViewTap" class="userinfo-avatar" src="{{userInfo.avatarUrl}}" mode="cover"></image>
        <text>{{userInfo.nickName}}</text>
      </block>
    </view>
    <view class="mt-[40px]">
      <text class="{{ motto }}">{{motto}}</text>
    </view>
  </view>
</scroll-view>