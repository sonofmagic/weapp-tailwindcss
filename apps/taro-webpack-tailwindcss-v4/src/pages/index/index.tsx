import { Text, View } from '@tarojs/components'
import { useLoad } from '@tarojs/taro'
import './index.css'

export default function Index() {
  useLoad(() => {
    console.log('Page loaded.')
  })

  return (
    <>
      <View className="bg-[#2e2bcc] text-[100rpx] text-white">
        <Text>Hello world!</Text>
      </View>
      <View className="bg-purple-800 text-pink-200">
        11
      </View>
      <View className="h-14 bg-gradient-to-r from-cyan-500 to-blue-500"></View>
      <View className="bg-purple-800 text-pink-200 invert">
        invert
      </View>
      <View className="bg-purple-800 text-pink-200 backdrop-blur-2xl">
        backdrop-blur-2xl
      </View>
      <View className="skew-x-0 bg-purple-800 text-pink-200">
        skew-x-0
      </View>
      <View className="flex space-x-1 space-y-1.5">
        <View className="bg-purple-800 text-pink-200">
          Tailwind
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind
        </View>
        <Text className="bg-purple-800 text-pink-200">
          Tailwind
        </Text>
      </View>
      <View className="flex flex-col space-y-1.5">
        <View className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </View>
        <Text className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </Text>
      </View>
    </>

  )
}
