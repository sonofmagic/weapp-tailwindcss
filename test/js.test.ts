/* eslint-disable no-template-curly-in-string */
import { getClassCacheSet } from 'tailwindcss-patch'
// import { createGetCase, jsCasePath } from './util'
import { createGetCase, jsCasePath, createPutCase } from './util'
import { SimpleMappingChars2String } from '@/escape'
import { createJsHandler } from '@/js/index'
import { getCss } from '#test/helpers/getTwCss'
import { getOptions } from '@/options'
import { defaultOptions } from '@/defaults'

const getCase = createGetCase(jsCasePath)
const putCase = createPutCase(jsCasePath)

const testTable = [
  {
    name: 'common'
  }
] as {
  name: string
  strategy?: 'replace'
}[]

describe('jsHandler', () => {
  let h: ReturnType<typeof createJsHandler>
  // let h: ReturnType<typeof createJsHandler>
  // let h: ReturnType<typeof createJsHandler>
  let dh: ReturnType<typeof createJsHandler>
  // let smh: ReturnType<typeof createJsHandler>
  let defaultJsHandler: ReturnType<typeof createJsHandler>
  beforeEach(() => {
    h = createJsHandler({
      escapeMap: SimpleMappingChars2String
    })

    // h = createJsHandler({
    //   escapeMap: SimpleMappingChars2String,
    //   minifiedJs: true
    // })

    dh = createJsHandler({
      escapeMap: SimpleMappingChars2String,
      // minifiedJs: true,
      arbitraryValues: {
        allowDoubleQuotes: true
      }
    })

    // smh = createJsHandler({
    //   escapeMap: SimpleMappingChars2String,
    //   generateMap: true
    // })

    const { jsHandler } = getOptions()
    defaultJsHandler = jsHandler
  })
  it.each(testTable)('$name common case', () => {
    const set: Set<string> = new Set()
    set.add('text-[12px]')
    set.add('flex')
    set.add('w-2.5')

    const code = h(`const n = 'text-[12px] flex bg-[red] w-2.5'`, set).code
    expect(code).toBe("const n = 'text-_12px_ flex bg-[red] w-2d5'")
  })

  it.each(testTable)('$name common case with ignore comment', () => {
    const set: Set<string> = new Set()
    set.add('text-[12px]')
    set.add('flex')
    set.add('w-2.5')

    const code = h(`const n = /*weapp-tw ignore*/ 'text-[12px] flex bg-[red] w-2.5'`, set).code
    expect(code).toBe("const n = /*weapp-tw ignore*/ 'text-[12px] flex bg-[red] w-2.5'")
  })

  it.each(testTable)('$name preserve space', () => {
    const set: Set<string> = new Set()
    set.add('text-[12px]')
    set.add('flex')
    set.add('w-2.5')
    const xxx = h

    const code = xxx("const n = 'text-[12px] flex \\n bg-[red] w-2.5'", set).code
    expect(code).toBe("const n = 'text-_12px_ flex \\n bg-[red] w-2d5'")
  })

  it.each(testTable)('$name preserve space case2', () => {
    const set: Set<string> = new Set()
    set.add('text-[12px]')
    set.add('flex')

    const code = h('const n = `text-[12px] \\n\\n  flex  \\n\\n  bg-[red]`', set).code
    expect(code).toBe('const n = `text-_12px_ \\n\\n  flex  \\n\\n  bg-[red]`')
  })

  it.each(testTable)('$name babel TemplateElement case', () => {
    const set: Set<string> = new Set()
    set.add('text-[12px]')
    set.add('flex')
    set.add('bg-[red]')

    const code = h("const p = 'text-[12px]';const n = `${p} \\n\\n  flex  \\n\\n  bg-[red] '`", set).code
    expect(code).toBe("const p = 'text-_12px_';const n = `${p} \\n\\n  flex  \\n\\n  bg-_red_ '`")
  })

  it.each(testTable)('$name TemplateElement case 0', () => {
    const set: Set<string> = new Set()
    set.add('text-[12px]')
    set.add('text-[199px]')
    set.add('flex')
    set.add("bg-[url('天气好')]")
    set.add('bg-[red]')

    const code = h("const p = 'text-[12px]';const n = `bg-[url('天气好')]${p}text-[199px] \\n\\n  flex  \\n\\n  bg-[red] '`", set).code
    expect(code).toBe("const p = 'text-_12px_';const n = `bg-_url_qu5929u6c14u597dq__${p}text-_199px_ \\n\\n  flex  \\n\\n  bg-_red_ '`")
  })

  it.each(testTable)('$name mpx jit classNames', () => {
    const testCase = `data: {
      classNames: "text-[#123456] text-[50px] bg-[#fff]"
    }`

    const set: Set<string> = new Set()
    set.add('text-[#123456]')
    set.add('bg-[#fff]')
    set.add('text-[50px]')

    const code = h(testCase, set).code
    expect(code).toMatchSnapshot()
  })

  it.each(testTable)('$name img url case', () => {
    const testCase = `data: {
      classNames: "bg-[url('https://ylnav.com/assets/images/vu/divider-gray.webp')]"
    }`

    const set: Set<string> = new Set()
    set.add("bg-[url('https://ylnav.com/assets/images/vu/divider-gray.webp')]")

    const code = h(testCase, set).code
    expect(code).toMatchSnapshot()
  })

  it('"after:content-["对酒当歌，人生几何"]"', async () => {
    const testCase = 'const a = \'after:content-["对酒当歌，人生几何"]\''
    await getCss(testCase)
    const set = getClassCacheSet()
    const code = dh(testCase, set).code
    expect(code).toMatchSnapshot()
  })

  it.each(testTable)('$name "after:content-[\'对酒当歌，人生几何\']"', async () => {
    const testCase = 'const a = "after:content-[\'对酒当歌，人生几何\']"'
    await getCss(testCase)
    const set = getClassCacheSet()

    const code = h(testCase, set).code
    expect(code).toMatchSnapshot()
  })

  it("jsPreserveClass '*' keyword", () => {
    const set: Set<string> = new Set()
    set.add('*')

    const code = defaultJsHandler("const n = '* 1 * 2'", set).code

    expect(code).toMatchSnapshot()
  })

  it("jsPreserveClass '*' and 'w-[100px]' keyword", () => {
    const set: Set<string> = new Set()
    set.add('*')
    set.add('w-[100px]')
    const myCustomJsHandler = createJsHandler({
      escapeMap: SimpleMappingChars2String,
      jsPreserveClass(keyword) {
        if (defaultOptions.jsPreserveClass?.(keyword)) {
          return true
        }
        if (keyword === 'w-[100px]') {
          return true
        }
      }
    })
    const code = myCustomJsHandler("const n = '* 1 * 2 w-[100px]'", set).code

    expect(code).toMatchSnapshot()
  })

  it("jsPreserveClass '*' and but not 'w-[100px]' keyword", () => {
    const set: Set<string> = new Set()
    set.add('*')
    set.add('w-[100px]')
    const myCustomJsHandler = createJsHandler({
      escapeMap: SimpleMappingChars2String,
      jsPreserveClass(keyword) {
        if (defaultOptions.jsPreserveClass?.(keyword)) {
          return true
        }
        // if (keyword === 'w-[100px]') {
        //   return true
        // }
      }
    })
    const code = myCustomJsHandler("const n = '* 1 * 2 w-[100px]'", set).code
    expect(code).toBe("const n = '* 1 * 2 w-_100px_'")
  })

  it("[replace] jsPreserveClass '*' and but not 'w-[100px]' keyword", () => {
    const set: Set<string> = new Set()
    set.add('*')
    set.add('w-[100px]')
    const myCustomJsHandler = createJsHandler({
      escapeMap: SimpleMappingChars2String,
      jsPreserveClass(keyword) {
        if (defaultOptions.jsPreserveClass?.(keyword)) {
          return true
        }
        // if (keyword === 'w-[100px]') {
        //   return true
        // }
      }
    })
    const code = myCustomJsHandler("const n = '* 1 * 2 w-[100px]'", set).code
    expect(code).toBe("const n = '* 1 * 2 w-_100px_'")
  })

  it('LINEFEED case', () => {
    const testCase = 'const LINEFEED = "\\n";'
    const set = new Set<string>()
    const code = h(testCase, set).code
    // 'const LINEFEED = "\n";'
    // 'const LINEFEED = "\\n";'
    expect(code).toBe(testCase)

    // \n 被展开导致的错误
    //     const LINEFEED = "
    // ";
  })

  it('mangleContext case', () => {
    const set: Set<string> = new Set()
    // set.add('*')
    set.add('w-[100px]')
    const { jsHandler, setMangleRuntimeSet } = getOptions({
      mangle: true
    })
    setMangleRuntimeSet(set)
    const code = jsHandler("const n = '* 1 * 2 w-[100px]'", set).code
    expect(code).toBe("const n = '* 1 * 2 tw-a'")
  })

  // it('source map case 0', () => {
  //   const set: Set<string> = new Set()
  //   set.add('w-[100px]')
  //   const { jsHandler, setMangleRuntimeSet } = getOptions()
  //   setMangleRuntimeSet(set)
  //   const { code, map } = jsHandler("const n = '* 1 * 2 w-[100px]'", set)
  //   expect(code).toBe("const n = '* 1 * 2 w-_100px_'")
  //   expect(map).toBeTruthy()
  //   expect(map?.toString()).toBe('{"version":3,"sources":[""],"names":[],"mappings":"AAAA,WAAW,iBAAiB"}')
  // })

  it('eval StringLiteral case 0', () => {
    const set: Set<string> = new Set()
    set.add('w-[100px]')
    const code = h(`eval("const cls = 'w-[100px]';console.log(cls)")`, set).code
    expect(code).toBe('eval("const cls = \\\'w-_100px_\\\';console.log(cls)")')
  })

  it('eval TemplateElement case 0', () => {
    const set: Set<string> = new Set()
    set.add('w-[100px]')
    const code = h("eval(`const cls = 'w-[100px]';console.log(cls)`)", set).code
    expect(code).toBe("eval(`const cls = 'w-_100px_';console.log(cls)`)")
  })

  it('eval StringLiteral case regen 0', () => {
    const set: Set<string> = new Set()
    set.add('w-[100px]')
    const code = h(`eval("const cls = 'w-[100px]';console.log(cls)")`, set).code
    expect(code).toBe('eval("const cls = \\\'w-_100px_\\\';console.log(cls)")')
  })

  it('eval TemplateElement case regen 0', () => {
    const set: Set<string> = new Set()
    set.add('w-[100px]')
    const code = h("eval(`const cls = 'w-[100px]';console.log(cls)`)", set).code
    expect(code).toBe("eval(`const cls = 'w-_100px_';console.log(cls)`)")
  })

  it('jsStringEscape getCase 0', async () => {
    const aarun = await getCase('jsStringEscape.js')
    const set: Set<string> = new Set()
    set.add('w-[100px]')
    set.add('w-[99px]')
    const code = h(aarun, set).code
    await putCase('jsStringEscape.res.js', code)
    expect(code).toMatchSnapshot()
  })
  // it('eval StringLiteral case 1', () => {
  //   const set: Set<string> = new Set()
  //   set.add('w-[100px]')
  //   const code = h(`eval("const cls = 'w-[100px]'\\\n;console.log(cls)")`, set).code
  //   expect(code).toBe('eval("const cls = \\\'w-_100px_\\\';console.log(cls)")')
  // })

  it('taro-url-before case', async () => {
    const aarun = await getCase('taro-url-before.js')
    const set: Set<string> = new Set()
    const code = h(aarun, set).code
    expect(code).toMatchSnapshot()
  })

  it('unicode case 0', async () => {
    const unicodeCase = await getCase('taro-url-unicode.js')
    const set: Set<string> = new Set()
    const code = h(unicodeCase, set).code
    expect(code).toMatchSnapshot()
  })

  it('taro-lottie-miniprogram-dev', async () => {
    const testCase = await getCase('taro-lottie-miniprogram-dev.js')
    const set: Set<string> = new Set()
    const { jsHandler } = getOptions()
    expect(() => {
      const { code } = jsHandler(testCase, set)
      expect(code).toBe(testCase)
    }).not.toThrow()
  })

  it('taro-lottie-miniprogram-build-no-compress', async () => {
    const testCase = await getCase('taro-lottie-miniprogram-build-no-compress.js')
    const set: Set<string> = new Set()
    const { jsHandler } = getOptions()
    expect(() => {
      const { code } = jsHandler(testCase, set)
      expect(code).toBe(testCase)
    }).not.toThrow()
  })

  it('taro-lottie-miniprogram-build', async () => {
    const testCase = await getCase('taro-lottie-miniprogram-build.js')
    const set: Set<string> = new Set()
    const { jsHandler } = getOptions()
    expect(() => {
      const { code } = jsHandler(testCase, set)
      expect(code).toBe(testCase)
    }).not.toThrow()
  })

  it('issues/276 case 0', () => {
    const testCase = `const x = "rounded-[20rpx] before:rounded-[20rpx]"`
    const set: Set<string> = new Set()
    set.add('rounded-[20rpx]')
    set.add('before:rounded-[20rpx]')
    const { jsHandler } = getOptions()
    const code = jsHandler(testCase, set)
    expect(code).toMatchSnapshot()
  })

  it('issues/276 case 1', () => {
    const testCase = `const x = "rounded-[20rpx] before:rounded-[20rpx]"`
    const set: Set<string> = new Set()
    set.add('before:rounded-[20rpx]')
    set.add('rounded-[20rpx]')

    const { jsHandler } = getOptions()
    const code = jsHandler(testCase, set)
    expect(code).toMatchSnapshot()
  })

  it('issues/276 case 2', () => {
    const testCase = `const x = "before:rounded-[20rpx] rounded-[20rpx]"`
    const set: Set<string> = new Set()
    set.add('rounded-[20rpx]')
    set.add('before:rounded-[20rpx]')
    const { jsHandler } = getOptions()
    const code = jsHandler(testCase, set)
    expect(code).toMatchSnapshot()
  })

  it('issues/276 case 3', () => {
    const testCase = `const x = "before:rounded-[20rpx] rounded-[20rpx]"`
    const set: Set<string> = new Set()
    set.add('before:rounded-[20rpx]')
    set.add('rounded-[20rpx]')

    const { jsHandler } = getOptions()
    const code = jsHandler(testCase, set)
    expect(code).toMatchSnapshot()
  })
})
