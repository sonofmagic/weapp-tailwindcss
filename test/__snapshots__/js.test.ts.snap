// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`jsHandler > "after:content-["对酒当歌，人生几何"]" 1`] = `"const a = 'afterccontent-_xu5bf9u9152u5f53u6b4cuff0cu4ebau751fu51e0u4f55x_'"`;

exports[`jsHandler > 'common' "after:content-['对酒当歌，人生几何']" 1`] = `"const a = "afterccontent-_qu5bf9u9152u5f53u6b4cuff0cu4ebau751fu51e0u4f55q_""`;

exports[`jsHandler > 'common' img url case 1`] = `
"data: {
      classNames: "bg-_url_qhttpscssylnavdcomsassetssimagessvusdivider-graydwebpq__"
    }"
`;

exports[`jsHandler > 'common' mpx jit classNames 1`] = `
"data: {
      classNames: "text-_h123456_ text-_50px_ bg-_hfff_"
    }"
`;

exports[`jsHandler > jsPreserveClass '*' and 'w-[100px]' keyword 1`] = `"const n = '* 1 * 2 w-[100px]'"`;

exports[`jsHandler > jsPreserveClass '*' keyword 1`] = `"const n = '* 1 * 2'"`;

exports[`jsHandler > jsStringEscape getCase 0 1`] = `
"const a = 'w-_100px_';

eval('const b = \\'w-_99px_\\';\\nconsole.log(b)');"
`;

exports[`jsHandler > taro-url-before case 1`] = `
"var findReg = /[!'()~]|%20|%00/g;
var plusReg = /\\+/g;
var replaceCharMap = {
  '!': '%21',
  "'": '%27',
  '(': '%28',
  ')': '%29',
  '~': '%7E',
  '%20': '+',
  '%00': '\\x00'
};
"
`;

exports[`jsHandler > unicode case 0 1`] = `
"var replaceCharMap = {
  "'": '%27',
  '%00': '\\x00'
};
"
`;
